machine_learning_hep/optimization.py

```diff
@@ -269,11 +279,14 @@ def study_signif(case, names, bin_lim, file_mc, file_data, df_mc_reco, df_ml_tes
         plt.errorbar(x_axis, signif_array, yerr=signif_err_array, alpha=0.3, label=f'{name}',
                      elinewidth=2.5, linewidth=4.0)
 
+    if bitmap_dict["std_analysis_on_off"]:
+        df_ml_test = filter_bit_df(df_ml_test, gen_dict["bitselvariable"],
+                                   bitmap_dict["std_analysis_on_off"])
```

Comment:
Also this should be rearranged: here you are selecting the candidates that are mc signal (because of line 231-232) and pass the std cuts, if I understand correctly. Therefore, an "already cut" dataframe is passed to `calc_efficiency(...)` while it expects to take a dataframe with mc signal and apply the cut afterwards.

Reply:
This should be moved to inside the calc_efficiency function, before the 'num_sel_cand = len(df_to_sel)' statement. Then it is ok right?

Reply:
Yes, then it should work I think. However, if you want to keep the compatibility with the other selection method you should also keep the query and add back `sel_signal_reco_std` in the parameter database. Also line 31 of `calc_efficiency` should be checked. 

In general, all the selections done in `optimization.py` should consist in a query (dummy if you have the bitmap) and a filtering based on a bitmap (dummy if you have the old sel_signal) in order to keep the compatibility with the old approach.